src/librustc_typeck/diagnostics.rs

```diff
@@ -4095,6 +4095,63 @@ assert_eq!(!Question::No, true);
 ```
 "##,
 
+E0609: r##"
+Attempted to access a non-existent field in a struct was performed.
+
+Erroneous code example:
+
+```compile_fail,E0609
+struct StructWithFields {
+    x: u32,
+}
+
+let s = StructWithFields { x: 0 };
+println!("{}", s.foo); // error: no field `foo` on type `StructWithFields`
+```
+
+To fix this error, check if you didn't misspell the field's name or that the
+field actually exist. Example:
+
+```
+struct StructWithFields {
+    x: u32,
+}
+
+let s = StructWithFields { x: 0 };
+println!("{}", s.x); // ok!
+```
+"##,
+
+E0610: r##"
+Attempted to access a field on a primitive type.
+
+Erroneous code example:
+
+```compile_fail,E0610
+let x: u32 = 0;
+println!("{}", x.foo); // error: `{interger}` is a primitive type, therefore
+                       //        doesn't have fields
+```
+
+Primitive types are the most basic types available in Rust and don't have
+fields. If you want to use fields, take a look to the structs. Example:
```

Comment:
I don't know how the user would accidentally access a field of a variable when coding. I would think the most likely reason would be they a) didn't know the datatype of the variable they used or b) accidentally used the wrong variable. I don't know if suggesting the user looks at structs will be helpful in them resolving this error. I think this extra description may be unnecessary.

In case you want to keep it, I'd reword the second sentence as it reads as being very informal. Maybe instead "To access data via named fields, struct types are used."?